source/Octopus.Tentacle/Services/FileTransfer/FileTransferService.cs

```diff
@@ -71,9 +76,7 @@ public UploadResult UploadFile(string remotePath, DataStream upload)
             fileSystem.EnsureDiskHasEnoughFreeSpace(parentDirectory, upload.Length);
 
             log.Trace("Copying uploaded data stream to: " + fullPath);
-#pragma warning disable CS0612
-            upload.Receiver().SaveTo(fullPath);
-#pragma warning restore CS0612
+            upload.Receiver().SaveToAsync(fullPath, CancellationToken.None).Wait();
```

Comment:
Should this not just await?

Reply:
Indeed. I fixed it in [this PR](https://github.com/OctopusDeploy/OctopusTentacle/pull/662)